lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/DeltaPackingUtils.java

```diff
@@ -308,12 +306,7 @@ private static void fillArray8(byte[] output, int outputOffset, int length, byte
      */
     private static void fillArray4(short[] output, int outputOffset, int length, short baseValue)
     {
-        Slice buffer = Slices.wrappedShortArray(output, outputOffset, length * SHORTS_IN_LONG);
-        checkArgument(output.length - outputOffset >= length * SHORTS_IN_LONG, "Trying to write values out of array bounds");
-        long value = fillLong(baseValue);
-        for (int i = 0; i < length * Long.BYTES; i += Long.BYTES) {
-            buffer.setLong(i, value);
-        }
+        Arrays.fill(output, outputOffset, outputOffset + length, baseValue);
```

Comment:
The idea here was to speed up filling values by repeating a long instead of a short.
This is similar to `Slice#fill` but for short instead of byte
Can we provide `Slice#fill(short value)` in io.airlift.slice ? Or are we getting rid of `Slice#fill(byte value)` too ?

Reply:
Does this specific change actually matter in a full benchmark?  I ask because I wrote some benchmarks for the Slice custom loops and found that the built in stuff is generally faster (or as fast).

Reply:
We don't have a full benchmark, but we have a JMH benchmark `BenchmarkShortColumnReader (bitWidth=0, encoding=DELTA_BINARY_PACKED)`. I ran it and confirmed that `Arrays#fill` runs as fast or faster than current code. I have a vague memory that it helped when this code was written, but there's no reason to keep it now.

Reply:
In never JVMs, I believe these methods now have custom vectorized forms.